views/js/pciCreator/dev/audioRecordingInteraction/runtime/audioRecordingInteraction.js

```diff
@@ -638,60 +646,99 @@ define([
-
-                    } else if (self.player.is('playing')) {
-                        self.stopPlayback();
-                    }
-                }
-            }.bind(stop));
-            stop.on('updatestate', function() {
-                if ((self.player.is('playing') && !self._isAutoPlayingBack)
-                    || self.recorder.is('recording')) {
-                    this.enable();
-                } else {
-                    this.disable();
+            if (this.config.isReviewMode !== true) {
+                // Record button
+                record = uiElements.controlFactory({
+                    id: 'record',
+                    label: this.getControlIcon('record'),
+                    container: this.$controlsContainer
+                });
+                record.on(
+                    'click',
+                    function () {
+                        if (this.is('enabled')) {
+                            self.startRecording();
+                        }
+                    }.bind(record)
+                );
+                record.on(
+                    'updatestate',
+                    function () {
+                        if (
+                            self.player.is('created') &&
+                            !self.recorder.is('recording') &&
+                            ((self.hasMediaStimulus() && self.mediaStimulusHasPlayed()) || !self.hasMediaStimulus()) &&
+                            !self.getRecording()
+                        ) {
+                            this.enable();
+                        } else {
+                            this.disable();
+                        }
+                    }.bind(record)
+                );
+                this.controls.record = record;
+
+                // Stop button
+                stop = uiElements.controlFactory({
+                    id: 'stop',
+                    label: this.getControlIcon('stop'),
+                    container: this.$controlsContainer
+                });
+                stop.on(
+                    'click',
+                    function () {
+                        if (this.is('enabled')) {
+                            if (self.recorder.is('recording')) {
+                                self.stopRecording();
+                            } else if (self.player.is('playing')) {
+                                self.stopPlayback();
+                            }
+                        }
+                    }.bind(stop)
+                );
+                stop.on(
+                    'updatestate',
+                    function () {
+                        if ((self.player.is('playing') && !self._isAutoPlayingBack) || self.recorder.is('recording')) {
+                            this.enable();
+                        } else {
+                            this.disable();
+                        }
+                    }.bind(stop)
+                );
+                this.controls.stop = stop;
```

Comment:
**nitpick:** This means we cannot stop the playback when the review mode is active.

**suggestion:** Make the stop button always available, split its management in 2 sections, one for the recording, another for the playback. Or eventually, since the recording cannon be started in review mode, just keep it outside of the condition.

Reply:
1. Migration added

2. Stop button added

3. auto start of recording in review mode is prevented